rest/rest-util/src/main/java/org/ow2/proactive/scheduling/util/http/HttpClientBuilder.java

```diff
@@ -0,0 +1,168 @@
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ * ################################################################
+ * $$PROACTIVE_INITIAL_DEV$$
+ */
+package org.ow2.proactive.scheduling.util.http;
+
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+
+public class HttpClientBuilder {
+
+    public static final String HTTPS_ALLOW_ANY_CERTIFICATE = "web.https.allow_any_certificate";
+
+    public static final String HTTPS_ALLOW_ANY_HOSTNAME = "web.https.allow_any_hostname";
+
+    private static final AllowAllTrustStrategy ACCEPT_ANY_CERTIFICATE_TRUST_STRATEGY = new AllowAllTrustStrategy();
+
+    private Boolean allowAnyHostname;
+
+    private Boolean allowAnyCertificate;
+
+    private int maxConnections;
+
+    private boolean useSystemProperties;
+
+    public HttpClientBuilder() {
+
+    }
+
+    public HttpClientBuilder allowAnyHostname(boolean acceptAnyHostname) {
+        this.allowAnyHostname = acceptAnyHostname;
+        return this;
+    }
+
+    public HttpClientBuilder allowAnyCertificate(boolean acceptAnyCertificate) {
+        this.allowAnyCertificate = acceptAnyCertificate;
+        return this;
+    }
+
+    public HttpClientBuilder insecure(boolean insecure) {
+        allowAnyHostname(insecure);
+        allowAnyCertificate(insecure);
+        return this;
+    }
+
+    public HttpClientBuilder maxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+
+    public HttpClientBuilder useSystemProperties() {
+        return useSystemProperties(true);
+    }
+
+    public HttpClientBuilder useSystemProperties(boolean useSystemProperties) {
+        this.useSystemProperties = useSystemProperties;
+        return this;
+    }
+
+    public CloseableHttpClient build() {
+        org.apache.http.impl.client.HttpClientBuilder httpClientBuilder =
+                createHttpClientBuilder();
+
+        boolean acceptAnyCertificate = false;
+        boolean acceptAnyHostname = false;
+
+        if (useSystemProperties) {
+            httpClientBuilder.useSystemProperties();
+
+            String property = System.getProperty(HTTPS_ALLOW_ANY_CERTIFICATE);
+            if (property != null && property.equalsIgnoreCase("true")) {
+                acceptAnyCertificate = true;
+            }
+
+            property = System.getProperty(HTTPS_ALLOW_ANY_HOSTNAME);
+            if (property != null && property.equalsIgnoreCase("true")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "true" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)